### PR TITLE
os.args() and passthru from CLI

### PR DESCRIPTION
**Pull request by edhemphill**
i'll try to update this PR with some tests.

### CODE REVIEW

**Review by myzie — COMMENTED** [review]
This is looking good. Just left a couple minor questions/comments.

File: os/os.go [hunk 1]
```diff
@@ -47,6 +47,8 @@ type FS interface {
 
 type OS interface {
 	FS
+	Args() []string
+	SetArgs(args []string)
```
[the remark this answers is not in the document]
Comment (edhemphill) [reply os/os.go:51]: let me look at that. I put SetArgs in so calling code using risor in an embedded fashion could set the args easily.

File: os/virtual.go [hunk 2]
```diff
@@ -158,9 +159,20 @@ func NewVirtualOS(ctx context.Context, opts ...Option) *VirtualOS {
 	for _, opt := range opts {
 		opt(vos)
 	}
+	vos.args = globalScriptargs
```
[the remark this answers is not in the document]
Comment (edhemphill) [reply os/virtual.go:162]: ok - got it - ill adjust for that pattern

**Comment by edhemphill** [comment]
updated. see what you think

File: os/os.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (myzie) [reply os/os.go]: Looks good

**Comment by myzie** [comment]
@edhemphill - this all looks good. Just this one little update to do:
https://github.com/risor-io/risor/pull/102/files#r1321470932

After that I think this is good to go. I'll give it a try just to sanity check.

**Review by edhemphill — COMMENTED** [review]
good catch

**Review by myzie — APPROVED** [review]
This looks good to me.

```
go run ./cmd/risor -c "os.args()" -- -foo --bar baz
[
  "-foo",
  "--bar",
  "baz"
]
```

One thing we should look at as a follow-up is what the shebang line should be now. What I had in there before doesn't seem to work, but I don't think that needs to hold up this PR.

Not working:

```
#!/usr/bin/env risor --

os.args()
```

**Comment by edhemphill** [comment]
yeah - i see what you're saying... will try to fix. It's not executing the script.

**Comment by edhemphill** [comment]
so in my testing this works:

(test.tm)
```
#!/usr/bin/env /home/ed/work/risor/cmd/risor/risor 

os.args()
```

As well as some other variations with and without `env`. The issue is the script arg of the file with the `#!` is added after the `--`  so it gets passed to the risor script. 

With the above, you can pass args to the risor script like:

```
 ./test.tm -- 1 2 3 

[
  "1",
  "2",
  "3"
]
```

To get around having to pass args behind a `--` you could have a wrapper script for risor. As an example:

```
#!/bin/bash

exec /home/ed/work/risor/cmd/risor/risor "${1}" -- "${@:2}"
```

```
$ ./test.tm 1 2 3 

[
  "1",
  "2",
  "3"
]
```

**Pull request merged**